examples/stable_diffusion_xl/gm/models/trainer_factory.py

```diff
@@ -98,3 +144,73 @@ def construct(self, x, *tokens):
         loss, _, overflow = self.ldm_with_loss_grad(x, noised_input, sigmas, w, concat, context, y)
 
         return loss, overflow
+
+
+class TrainOneStepCellDreamBooth(nn.Cell):
```

Comment:
I feel like creating `TrainOneStepCell` for each model is suboptimal. Maybe you can refer to `examples.stable_diffusion_v2.ldm.modules.train.trainer.TrainOneStepWrapper` to see how we utilize `TrainOneStepCell` for all SD variations.

Reply:
marked as todo